src/cephadm/cephadm

```diff
@@ -4608,6 +4609,336 @@ def command_install():
 
 ##################################
 
+def get_ipv4_address(ifname):
+    # type: (str) -> str
+    def _extract(sock, offset):
+        return socket.inet_ntop(
+                socket.AF_INET,
+                fcntl.ioctl(
+                    sock.fileno(),
+                    offset,
+                    struct.pack('256s', bytes(ifname[:15], 'utf-8'))
+                )[20:24])
+        
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        addr = _extract(s, 35093)  # '0x8915' = SIOCGIFADDR
+        dq_mask = _extract(s, 35099)  # 0x891b = SIOCGIFNETMASK
+    except OSError:
+        # interface does not have an ipv4 address
+        return ''
+    
+    dec_mask = sum([bin(int(i)).count('1')
+                    for i in dq_mask.split('.')])
+    return '{}/{}'.format(addr, dec_mask)
+
+
+def get_ipv6_address(ifname):
+    # type: (str) -> str
+    if not os.path.exists('/proc/net/if_inet6'):
+        return ''
+
+    raw = rfiles(['/proc/net/if_inet6'])
+    data = raw.splitlines()
+    # based on docs @ https://www.tldp.org/HOWTO/Linux+IPv6-HOWTO/ch11s04.html
+    for iface_setting in data:
+        field = iface_setting.split()
+        if field[-1] == ifname:
+            return "{}/{}".format(field[0], int('0x{}'.format(field[2]), 16))
+    return ''
```

Comment:
Hm. We already have

https://github.com/ceph/ceph/blob/4c52403367cfbda8217ddc985c712387040d444b/src/cephadm/cephadm#L3332-L3363

Do we really need both?

Reply:
list_networks is providing the IP and subnet for all devices - what my functions do is provide ip for a given interface, so the returning daa is more granular. The idea being that orch or the UI could sanity check the network config. For example; is the subnet set up on the IPoIB interface, is the NIC's MTU correct, are all hosts using that subnet consistently configured (i.e. all use a NIC with the same MTU and see the same speed)

Reply:
Maybe the other way around works? We need to networks in order to guess the MON network settings. Can we get the same information from your functions?

Reply:
you could gather_facts at the early stages of bootstrap. This would give you the IP's available, with NIC up/down state, mtu etc ahead of the deployment - to you could do a pre-flight before bootstrap actually starts.